### PR TITLE
feat: Support PodDisruptionBudgets

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/435

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/zookeeper/pages/usage_guide/operations/pod-disruptions.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+fn quorum_size(num_servers: u16) -> u16 {
+    // Same as max((num_servers / 2) + 1, 1), but without the need for floating point arithmetics,
+    // which are subject to rounding errors.
+    max((num_servers + 2) / 2, 1)
+}
+
+// Minimum required amount of servers to form quorum.
+let quorum_size = quorum_size(num_servers);
+
+// Subtract once to not cause a single point of failure
```
Comment (maltesander) [inline docs/modules/zookeeper/pages/usage_guide/operations/pod-disruptions.adoc:26]:
```suggestion
// Subtract one to not cause a single point of failure
```

File: docs/modules/zookeeper/pages/usage_guide/operations/pod-disruptions.adoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+let max_unavailable = num_servers.saturating_sub(quorum_size).saturating_sub(1);
+
+// Clamp to at least a single node allowed to be offline, so we don't block Kubernetes nodes from draining.
+let max_unavailable = max(max_unavailable, 1)
+----
+
+This results e.g. in the following numbers:
+
+TIP: It is strongly recommended to use a odd number of servers (e.g. 3, 5 or 7).
```
Comment (maltesander) [inline docs/modules/zookeeper/pages/usage_guide/operations/pod-disruptions.adoc:35]:
```suggestion
TIP: It is strongly recommended to use an odd number of servers (e.g. 3, 5 or 7).
```

File: rust/operator-binary/src/operations/pdb.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+use crate::zk_controller::ZK_CONTROLLER_NAME;
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("Cannot create PodDisruptionBudget for role [{role}]"))]
+    CreatePdb {
+        source: stackable_operator::error::Error,
+        role: String,
+    },
+    #[snafu(display("Cannot apply role group PodDisruptionBudget [{name}]"))]
```
Comment (maltesander) [inline rust/operator-binary/src/operations/pdb.rs:19]: Not role-group right?
```suggestion
    #[snafu(display("Cannot apply PodDisruptionBudget [{name}]"))]
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things, LGTM otherwise!

File: rust/operator-binary/src/operations/pdb.rs (hunk 3 above)
Comment (sbernauer) [reply rust/operator-binary/src/operations/pdb.rs]: yes

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by fhennig** [comment]
@sbernauer check the checkmarks in the _Author_ list ;) follow the process!

**Comment by sbernauer** [comment]
done :)

**Comment by maltesander** [comment]
@fhennig is this the ticket police ;) ?

**Pull request merged**